core/version/src/github.rs

```diff
@@ -0,0 +1,94 @@
+use anyhow::anyhow;
+use metrics::counter;
+use self_update::backends::github::UpdateBuilder;
+
+use ya_core_model::version::Release;
+use ya_persistence::executor::DbExecutor;
+
+use crate::db::dao::ReleaseDAO;
+use crate::db::model::DBRelease;
+use crate::service::cli::ReleaseMessage;
+use std::convert::TryFrom;
+
+const REPO_OWNER: &'static str = "golemfactory";
+const REPO_NAME: &'static str = "yagna";
+
+pub async fn check_latest_release(db: &DbExecutor) -> anyhow::Result<Release> {
+    log::trace!("Checking latest Yagna release");
+    let gh_rel = UpdateBuilder::new()
+        .repo_owner(REPO_OWNER)
+        .repo_name(REPO_NAME)
+        .bin_name("") // seems required by builder but unused
+        .current_version("") // similar as above
+        .build()?
+        .get_latest_release()?;
+
+    let db_rel = DBRelease::try_from(gh_rel)?;
+    let rel = Release::from(db_rel.clone());
+    log::trace!("Got latest Yagna release {}", rel);
+    if self_update::version::bump_is_greater(ya_compile_time_utils::semver_str(), &rel.version)
+        .map_err(|e| {
+            anyhow!(
+                "Github release version `{}` parse error: {}",
+                rel.version,
+                e
+            )
+        })?
+    {
+        match db.as_dao::<ReleaseDAO>().save(db_rel).await {
+            Err(e) => log::error!("Storing new Yagna release {} to DB. {}", rel, e),
+            Ok(r) => {
+                counter!("version.new", 1);
+                log::warn!("{}", ReleaseMessage::Available(&r));
```

Comment:
I don't like the idea of claiming `pinger` responsibility here. It can result in ugly spamming of log more frequently than defined by interval in `pinger`.

Reply:
this will be invoked:
- every 24h by `worker`
- on every start
- manually via `yagna version check` 
It will **immediately** inform the user (via logs) that there is new version. This immediate notification was my goal here.

The same message will be logged by `pinger` every 30mins.

Reply:
So every 24h we will get double logs from pinger and worker?